pico/processor/src/main/java/io/helidon/pico/processor/ActiveProcessorUtils.java

```diff
@@ -0,0 +1,308 @@
+
+import io.helidon.builder.processor.spi.TypeInfoCreatorProvider;
+import io.helidon.builder.processor.tools.BuilderTypeTools;
+import io.helidon.common.HelidonServiceLoader;
+import io.helidon.common.types.TypeInfo;
+import io.helidon.common.types.TypedElementName;
+import io.helidon.pico.tools.Messager;
+import io.helidon.pico.tools.ModuleInfoDescriptor;
+import io.helidon.pico.tools.Options;
+import io.helidon.pico.tools.ServicesToProcess;
+
+import static io.helidon.pico.processor.GeneralProcessorUtils.toPath;
+import static io.helidon.pico.tools.ModuleUtils.REAL_MODULE_INFO_JAVA_NAME;
+import static io.helidon.pico.tools.ModuleUtils.inferSourceOrTest;
+
+/**
+ * Carries methods that are relative only during active APT processing.
+ *
+ * @see GeneralProcessorUtils
+ */
+final class ActiveProcessorUtils implements Messager {
+    static final boolean MAYBE_ANNOTATIONS_CLAIMED_BY_THIS_PROCESSOR = false;
+    static final String TARGET_DIR = "/target/";
+    static final String SRC_MAIN_JAVA_DIR = "/src/main/java";
+
+    private final System.Logger logger;
+    private final ProcessingEnvironment processingEnv;
+    private final TypeInfoCreatorProvider typeInfoCreatorProvider;
+    private RoundEnvironment roundEnv;
+
+    ActiveProcessorUtils(AbstractProcessor processor,
+                         ProcessingEnvironment processingEnv,
+                         RoundEnvironment roundEnv) {
+        this.logger = System.getLogger(processor.getClass().getName());
+        this.processingEnv = Objects.requireNonNull(processingEnv);
+        this.roundEnv = roundEnv;
+        this.typeInfoCreatorProvider = HelidonServiceLoader.create(
+                        ServiceLoader.load(TypeInfoCreatorProvider.class, TypeInfoCreatorProvider.class.getClassLoader()))
+                .asList()
+                .stream()
+                .findFirst()
+                .orElseThrow(() -> new IllegalStateException("No available " + TypeInfoCreatorProvider.class));
+
+        Options.init(processingEnv);
+        debug("*** Processing " + processor.getClass().getSimpleName() + " ***");
+    }
+
+    @Override
+    public void debug(String message,
+                      Throwable t) {
+        if (Options.isOptionEnabled(Options.TAG_DEBUG)) {
+            if (logger.isLoggable(loggerLevel())) {
+                logger.log(loggerLevel(), getClass().getSimpleName() + ": Debug: " + message, t);
+            }
+        }
+
+        if (processingEnv != null && processingEnv.getMessager() != null) {
+            processingEnv.getMessager().printMessage(Diagnostic.Kind.OTHER, message);
+        }
+    }
+
+    @Override
+    public void debug(String message) {
+        if (Options.isOptionEnabled(Options.TAG_DEBUG)) {
+            if (logger.isLoggable(loggerLevel())) {
+                logger.log(loggerLevel(), getClass().getSimpleName() + ": Debug: " + message);
+            }
+        }
+
+        if (processingEnv != null && processingEnv.getMessager() != null) {
```

Comment:
This code is repeated over and over again. Please replace it with something better.

Reply:
addressed